src/Internal/Types/BigNum.purs

```diff
@@ -116,6 +120,17 @@ fromString = _fromString maybeFfiHelper
 fromStringUnsafe :: String -> BigNum
 fromStringUnsafe = unsafePartial fromJust <<< fromString
 
+foreign import data StringOrNumber :: Type
+
+foreign import _fromStringOrNumber
+  :: MaybeFfiHelper -> StringOrNumber -> Maybe BigNum
+
+fromStringOrNumber :: StringOrNumber -> Maybe BigNum
+fromStringOrNumber = _fromStringOrNumber maybeFfiHelper
+
+stringToStringOrNumber :: String -> StringOrNumber
```

Comment:
The point is that CIP-30 standard uses String or Number to represent BigNumber amount argument.

After writing that code I found out that we use untagged union lib. I think it may be suited better than writing separate type?

Reply:
Also this is strange to me, why CIP-30 uses BigNumber instead of BigInteger here.

Reply:
It can't use BigNumber. It's CSL internal type. Maybe a typo in the spec.

Reply:
I see, they don't accept a BigNumber, they mean that the representation of the number in the string is the same as BigNumber string representation

Reply:
Please remove this type and just use String in the CIP-30 FFI code. The interface should accept a `Coin`.

Reply:
Sorry, I mixed up, I meant BigNum, not BigNumber.